public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -697,6 +702,19 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
       this._scrollRef?.scrollTo(0, this._prevScrollPos!);
     }
   }
+
+  private triggerRefreshEvent = ({ payload }: SceneObjectStateChangedEvent) => {
+    const isAnnotationChange =
+      payload.changedObject instanceof DashboardAnnotationsDataLayer &&
+      !Object.prototype.hasOwnProperty.call(payload.partialUpdate, 'data');
+    const isSceneVariableChange = isSceneVariableInstance(payload.changedObject);
+
+    if (isAnnotationChange || isSceneVariableChange) {
+      // DashboardModel triggers RefreshEvent when variables change
+      // To avoid compatibility issues in plugins we need to trigger the same event
+      this.publishEvent(new RefreshEvent());
+    }
+  };
```

Comment:
confused, we SceneTimeRange already publish this event https://github.com/grafana/scenes/blob/main/packages/scenes/src/core/SceneTimeRange.tsx#L187

Reply:
this would also emit way too many events for variable changes  (and for annotations)